src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,109 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    private int indexOfLastAddedElement = 0;
+    private Object[] array = new Object[10];
+
     @Override
     public void add(T value) {
-
+        if (indexOfLastAddedElement >= array.length) {
```

Comment:
if should be extracted to the method `checkSIze()` or kinda this and reuse it in all places